apps/emqx/src/emqx_config.erl

```diff
@@ -238,7 +247,22 @@ erase(RootName) ->
     persistent_term:erase(?PERSIS_KEY(?RAW_CONF, bin(RootName))).
 
 -spec put(emqx_map_lib:config_key_path(), term()) -> ok.
-put(KeyPath, Config) -> do_put(?CONF, KeyPath, Config).
+put(KeyPath, Config) ->
+    Putter = fun(Path, Map, Value) ->
+        emqx_map_lib:deep_put(Path, Map, Value)
+    end,
+    do_put(?CONF, safe, Putter, KeyPath, Config).
+
+-spec force_put(emqx_map_lib:config_key_path(), term()) -> ok.
+force_put(KeyPath, Config) ->
+    force_put(KeyPath, Config, safe).
+
+-spec force_put(emqx_map_lib:config_key_path(), term(), safe | unsafe) -> ok.
+force_put(KeyPath, Config, Safety) ->
```

Comment:
if we force convert unsafe atoms here, there is no need for the `Saftey` arg passed down to the lower level functions ?

Reply:
It goes down to the `do_deep_put` function, for which the safety is required